### PR TITLE
End to End Scenario for System Admin role

### PR DESCRIPTION
**Pull request by san7ket**
```
pytest tests/foreman/cli/test_role.py::SystemAdminTestCases::test_system_admin_role_end_to_end
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.6.0, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/sjagtap/PycharmProjects/robottelo/tests/foreman, inifile: pytest.ini
plugins: services-1.2.1, mock-1.6.3
collecting 1 item                                                                                                                                                                                                 2018-12-17 19:54:55 - conftest - DEBUG - Collected 1 test cases

collected 1 item                                                                                                                                                                                                  

tests/foreman/cli/test_role.py .                                                                                                                                                                            [100%]

=========================================================================================== 1 passed in 157.81 seconds ============================================================================================
```

### CODE REVIEW

**Comment by san7ket** [comment]
@xprazak2 can you please take a look at the doc-strings once again

**Comment by xprazak2** [comment]
@san7ket, the doc-strings look ok to me.

File: tests/foreman/cli/test_role.py [hunk 1]
```diff
@@ -1265,3 +1273,124 @@ def test_negative_assign_org_admin_to_ldap_user_group(self):
...
+            username=system_admin_1['login'],
+            password=common_pass).set({
+                'name': "outofsync_interval",
+                'value': "32"
+                })
+        sync_time = Settings.list({
+            'search': 'name=outofsync_interval'
+            })[0]
+        # Asserts if the setting was updated successfully
+        self.assertEqual('32', sync_time['value'])
```
Comment (ntkathole) [inline tests/foreman/cli/test_role.py]: @san7ket Can you please add logic to revert the change as settings are global? There may be chances that it will affect other tests.

File: tests/foreman/cli/test_role.py [hunk 2]
```diff
@@ -1265,3 +1273,124 @@ def test_negative_assign_org_admin_to_ldap_user_group(self):
...
+                    })
+        # Assert if the cloning was successful
+        self.assertIsNotNone(org_role['id'])
+        org_role_filters = Role.filters({'id': org_role['id']})
+        search_filter = None
+        for arch_filter in org_role_filters:
+            if arch_filter['resource-type'] == 'Architecture':
+                search_filter = arch_filter
+                break
+        print(search_filter)
```
Comment (ntkathole) [inline tests/foreman/cli/test_role.py]: print ?? do we need it?

Comment (san7ket) [reply tests/foreman/cli/test_role.py]: ahh , debug statement

**Comment by san7ket** [comment]
Added teardown to revert the changed settings

```
 pytest tests/foreman/cli/test_role.py::SystemAdminTestCases::test_system_admin_role_end_to_end
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.6.0, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/sjagtap/PycharmProjects/robottelo/tests/foreman, inifile: pytest.ini
plugins: xdist-1.25.0, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.6.0
collecting 1 item                                                                                                                                                                                                 2018-12-21 12:25:50 - conftest - DEBUG - Collected 1 test cases

collected 1 item                                                                                                                                                                                                  

tests/foreman/cli/test_role.py .                                                                                                                                                                            [100%]

=========================================================================================== 1 passed in 171.07 seconds ============================================================================================



2018-12-21 12:28:34 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f127461c5c0
2018-12-21 12:28:37 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f12745ff470
2018-12-21 12:28:37 - robottelo.ssh - INFO - Connected to []
2018-12-21 12:28:37 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme  settings set --name="outofsync_interval" --value="30"'
2018-12-21 12:28:40 - robottelo.ssh - INFO - <<< stdout
Setting [outofsync_interval] updated to [30].
```
